### PR TITLE
Feature/32

### PR DESCRIPTION
**Pull request by ki-arie**
Lot's to unpack here.

Happy for code walkthrough before the merge.

### CODE REVIEW

File: native_client/modelstate.cc [hunk 1]
```diff
@@ -50,10 +50,9 @@ ModelState::decode_metadata(const DecoderState& state,
 
     for (int j = 0; j < out[i].tokens.size(); ++j) {
       TokenMetadata token {
-        strdup(alphabet_.DecodeSingle(out[i].tokens[j]).c_str()),      // text
+        strdup(alphabet_.DecodeSingle(out[i].tokens[j]).c_str()),   // text
         static_cast<unsigned int>(out[i].timesteps[j]),                // timestep
-        out[i].timesteps[j] * ((float)audio_win_step_ / sample_rate_), // start_time 
-        static_cast<float>(std::exp(out[i].probabilities[j]))          // probability
```
Comment (ki-arie) [inline native_client/modelstate.cc:56]: FYI @utunga this is the line where we add an extra exp to the logit`static_cast<float>(std::exp(out[i].probabilities[j]))`.

I'm taking it out atm, but we will add it using another ticket/feature, just to make sure it's the only functionality we're testing/adding

Comment (utunga) [reply native_client/modelstate.cc:56]: Hmm.. on further consideration, I dunno. 

Taking it out completely is a breaking change.

Leaving it with ```std::exp()``` is (almost certainly) incorrect.

Given that by far the bulk of the hassle here is the whole business of building and tagging the binaries and getting them deployed I'm erring on the side of suggesting we just put it (back) to ... 

```
static_cast<float>(out[i].probabilities[j]) // probability
```

That way the per char metadata will at least be there. We can still treat it as a separate task to decide if this actually helps or hinders matters.

Comment (ki-arie) [reply native_client/modelstate.cc:56]: Oh man, just realized this PR overwrites some changes you did on the API to expose the char confidences, I'll correct this up tmr

**Comment by utunga** [comment]
Just a quick comment on status of where I'm at.. have most of the tooling installed I think but with bazel 3.1.0 set as default and a clean check out feature/32 and submodules etc I am currently getting this 

image

`INFO: From Linking external/ruy/ruy/profiler/libinstrumentation.a:
warning: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning for library: bazel-out/darwin-opt/bin/external/ruy/ruy/profiler/libinstrumentation.a the table of contents is empty (no object file members in the library define global symbols)
ERROR: /Users/utunga/papa/DeepSpeech/tensorflow/native_client/BUILD:114:1: C++ compilation of rule '//native_client:sentencefit_bundle' failed (Exit 1)
native_client/sentencefit.cc:7:10: fatal error: 'xtensor/xarray.hpp' file not found
#include "xtensor/xarray.hpp"
         ^~~~~~~~~~~~~~~~~~~~
1 error generated.`

On this occasion I chose all defaults when running ./configure in tensorflow dir (which includes not enabling suport for iOS)

**Comment by utunga** [comment]
PS. I used `git pull --recurse-submodules
`
And it was not succesful 

So then.I tried 
`git submodule update --init --recursive`

And it was super effective. 

Also, forget the ./configure step. I guess that's not needed. 

PS I guess the problem is that I did 
git clone --recursive 

but then I checked out the branch and it didn't update the submodules. Even, as I say after git pull --recurse. But yeah update --init seems to be the go. ```

```
INFO: Elapsed time: 354.495s, Critical Path: 130.95s
INFO: 1228 processes: 1228 local.
INFO: Build completed successfully, 1229 total actions
```

File: native_client/python/impl.i [hunk 2]
```diff
@@ -74,6 +79,34 @@ static PyObject *parent_reference() {
...
+  def __repr__(self):
+    tokens_repr = ',\n'.join(repr(i) for i in self.tokens)
+    tokens_repr = '\n'.join('  ' + l for l in tokens_repr.split('\n'))
+    return 'SentenceFit tokens=[\n{}\n])'.format(tokens_repr)
+%}
+}
+
 %extend struct TokenMetadata {
 %pythoncode %{
   def __repr__(self):
```
Comment (utunga) [inline native_client/python/impl.i:112]: request that we add the prob to the default print output

**Review by utunga — CHANGES_REQUESTED** [review]
OK so we were going to add that tiny bit of print data and then just verify that we're getting probability metadata in a/the test

**Review by utunga — CHANGES_REQUESTED** [review]
Eek sorry doing this again.. not sure if this will show up twice now. 

Requested changes
-- Add probablity to impl.i print output for Metadata
-- Remove the exp() around probability in the output step (do'h) 
-- Verify that we're getting metadata in the test

File: native_client/python/make_test_python.sh [hunk 3]
```diff
@@ -0,0 +1,5 @@
+make bindings-clean
+pip uninstall deepspeech -y
+make bindings
+pip install dist/*.whl
```
Comment (utunga) [inline native_client/python/make_test_python.sh:4]: 👍

**Review by utunga — APPROVED** [review]
AWESOME 

WFM

**Pull request merged**